Client/Application/Delegates/SceneDelegate.swift

```diff
@@ -55,39 +78,436 @@ class SceneDelegate: UIResponder, UIWindowSceneDelegate {
+            }
+        } else {
+            urp.pingIfEnoughTimePassed()
+            self.backgroundDataSource.startFetching()
+        }
+    }
+}
+
+extension UIWindowScene {
+    /// A single scene should only have ONE browserViewController
+    /// However, it is possible that someone can create multiple,
+    /// Therefore, we support this possibility if needed
+    var browserViewControllers: [BrowserViewController] {
+        windows.compactMap({
+            $0.rootViewController as? UINavigationController
+        }).flatMap({
+            $0.viewControllers.compactMap({
+                $0 as? BrowserViewController
+            })
+        })
+    }
+    
+    /// A scene should only ever have one browserViewController
+    /// Returns the first instance of `BrowserViewController` that is found in the current scene
+    var browserViewController: BrowserViewController? {
+        return browserViewControllers.first
+    }
+}
+
+extension UIView {
+    /// Returns the `Scene` that this view belongs to.
+    /// If the view does not belong to a scene, it returns the scene of its parent
+    /// Otherwise returns nil if no scene is associated with this view.
+    var currentScene: UIWindowScene? {
+        if let scene = window?.windowScene {
+            return scene
+        }
+        
+        if let scene = superview?.currentScene {
+            return scene
+        }
+        
+        return nil
+    }
+}
+
+extension UIViewController {
+    /// Returns the `Scene` that this controller belongs to.
+    /// If the controller does not belong to a scene, it returns the scene of its presenter or parent.
+    /// Otherwise returns nil if no scene is associated with this controller.
+    var currentScene: UIWindowScene? {
+        if let scene = view.window?.windowScene {
+            return scene
+        }
+        
+        if let scene = parent?.currentScene {
+            return scene
+        }
+        
+        if let scene = presentingViewController?.currentScene {
+            return scene
+        }
+        
+        return nil
+    }
+}
+
+extension UIApplication {
+    /// An application can have multiple UIScene and UISceneSession
+    /// Returns ALL `BrowserViewController` from EVERY scene.
+    var browserViewControllers: [BrowserViewController] {
+        return UIApplication.shared.connectedScenes.compactMap({
+            $0 as? UIWindowScene
+        }).flatMap({
+            $0.browserViewControllers
+        })
+    }
+    
+    /// Should NEVER be used
+    /// Returns the very first `BrowserViewController` found in any `UIScene`
+    /// within the entire application. If this function is being used, it is time to refactor your code.
+    var browserViewController: BrowserViewController? {
+        log.debug("********* WARNING *********: Accessing `browserViewController` on `UIApplication`!!!\n DOES NOT SUPPORT MULTIPLE WINDOWS.")
+        
+        return browserViewControllers.first
```

Comment:
Is this used now? If not, can we delete it?

Reply:
It is used in 2 places with `UIApplication.shared.browserViewController`. These two places I have zero idea how to refactor as they're static content scripts and singletons :(